tests/integration/nm/linux_bridge_test.py

```diff
@@ -313,3 +313,85 @@ def test_bridge_enslave_unmanaged_interface(nm_unmanaged_dummy1):
             }
         )
         libnmstate.apply(desired_state)
+
+
+@pytest.fixture
+def nm_down_unmanaged_dummy1():
+    cmdlib.exec_cmd(
+        f"ip link add name {DUMMY1} type dummy".split(" "), check=True
+    )
+    yield
+    cmdlib.exec_cmd(
+        f"nmcli dev set {DUMMY1} managed yes".split(" "), check=True
+    )
+    time.sleep(1)  # Wait device became managed
```

Comment:
in a follow up PR: cannot we wait for an event? sleep() is inherently raceful.

Reply:
This is just for testing. On Nmstate implementation it is working with events/signals and async. calls. But on the integration test we are just invoking a shell command.

Reply:
thanks for the clarification. In a follow-up PR, please consider adding a similar logic to the tests. You don't want the test to be needlessly slow (why wait for 1 second, if NM did its thing in 100mS?) and you don't want the test to succeed just because NM was too slow to set the device to managed. The fact that you need a comment to explain the `sleep()` hints that something like `wait_for_device_is_managed(dummy1, timeout=1)` is in place.

Reply:
Sure, that is a nice improvement. I will add this to the idea lists we do have.